### PR TITLE
Initial random devices implementation

### PR DESCRIPTION
**Pull request by igor725**

### CODE REVIEW

File: core/fileManager/types/type_dev.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "type_dev.h"
+
+#include "type_random.h"
+
+std::unique_ptr<TypeDev> createType_dev(std::filesystem::path path, std::ios_base::openmode mode) {
+  // todo: /dev/rng?
+  if (path == "/dev/urandom" || path == "/dev/urandom") {
+    return createType_random();
+  } else { // todo: other devices
+    throw std::runtime_error("Unknown device!");
```
Comment (SysRay) [inline core/fileManager/types/type_dev.cpp:10]: Log Crit with todo pls. Windows and Linux exceptions don't mix well.

Comment (SysRay) [reply core/fileManager/types/type_dev.cpp:10]: Or catch before going back to the linux space

File: core/fileManager/types/type_dev.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "../ifile.h"
+
+#include <filesystem>
+#include <memory>
+
+class TypeDev: public IFile {
```
Comment (SysRay) [inline core/fileManager/types/type_dev.h:8]: Place the class inside the cpp. No need to make it public.

Comment (SysRay) [inline core/fileManager/types/type_dev.h:8]: Why another interface ?

Comment (igor725) [reply core/fileManager/types/type_dev.h:8]: I thought it will be easier to implement other devices with ITypeDev

Comment (SysRay) [reply core/fileManager/types/type_dev.h:8]: It just forwards the parent. If needed, we can just create an interface and the childs derive both

**Pull request merged**